app/actions/views/channel.js

```diff
@@ -273,16 +274,63 @@ export function handleSelectChannel(channelId) {
     };
 }
 
-export function handlePostDraftChanged(channelId, postDraft) {
+export function handlePostDraftChanged(channelId, draft) {
     return async (dispatch, getState) => {
         dispatch({
             type: ViewTypes.POST_DRAFT_CHANGED,
             channelId,
-            postDraft
+            draft
         }, getState);
     };
 }
 
+export function handlePostDraftSelectionChanged(channelId, cursorPosition) {
+    return {
+        type: ViewTypes.POST_DRAFT_SELECTION_CHANGED,
+        channelId,
+        cursorPosition
+    };
+}
+
+export function insertToDraft(value) {
+    return (dispatch, getState) => {
+        const state = getState();
+        const channelId = getCurrentChannelId(state);
+        const threadId = state.views.thread.currentThreadId;
+
+        let draft;
+        let cursorPosition;
+        let action;
+        if (threadId) {
+            draft = state.views.thread.drafts[threadId].draft;
+            cursorPosition = state.views.thread.drafts[threadId].cursorPosition;
```

Comment:
maybe you should get `state.views.thread.drafts[threadId]` and then access the `draft` and the `cursorPosition` and make sure that object actually exists